src/iuse.cpp

```diff
@@ -3164,13 +3164,29 @@ int iuse::makemound(player *p, item *it, bool, const tripoint& )
     }
 }
 
-int iuse::dig(player *p, item *it, bool, const tripoint & )
+int iuse::dig(player *p, item *it, bool, const tripoint &pos )
 {
-    for( const tripoint &pt : closest_tripoints_first( 1, p->pos() ) ) {
+    for( const tripoint &pt : closest_tripoints_first( 1, pos ) ) {
         if( g->m.furn_at( pt ).examine == iexamine::rubble ) {
-            p->moves -= 200;
             p->add_msg_if_player( _("You clear up that %s."), g->m.furnname( pt ).c_str() );
             g->m.furn_set( pt, f_null );
+
+            // costs per tile:
+            // DIG 2 = 300 seconds, 10 fatigue, hunger and thirst
+            // DIG 3 =  75 seconds,  2 fatigue, hunger and thirst
+            // DIG 4 =  33 seconds,  1 fatigue, hunger and thirst
+            // DIG 5 =  18 seconds,  0 fatigue, hunger and thirst
+            int bonus = std::max( it->get_quality( quality_id( "DIG" ) ) - 1, 1 );
+            bonus *= bonus;
+
+            p->moves -= 5000 / ( bonus * bonus );
+
+            if( p ) {
+                p->mod_fatigue( 10 / bonus );
```

Comment:
Fatigue means sleepiness, not physical fatigue. Shoveling away at rubble shouldn't put you to sleep.